cocos/base/CCStencilBitsManager.hpp

```diff
@@ -0,0 +1,61 @@
+/****************************************************************************
+ Copyright (c) 2010-2012 cocos2d-x.org
+ Copyright (c) 2013-2015 Chukong Technologies Inc.
+ 
+ http://www.cocos2d-x.org
+ 
+ Permission is hereby granted, free of charge, to any person obtaining a copy
+ of this software and associated documentation files (the "Software"), to deal
+ in the Software without restriction, including without limitation the rights
+ to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ copies of the Software, and to permit persons to whom the Software is
+ furnished to do so, subject to the following conditions:
+ 
+ The above copyright notice and this permission notice shall be included in
+ all copies or substantial portions of the Software.
+ 
+ THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ THE SOFTWARE.
+ ****************************************************************************/
+#ifndef StencilBitsManager_hpp
+#define StencilBitsManager_hpp
+#include "base/ccConfig.h"
+#include "platform/CCPlatformMacros.h"
+#include "platform/CCGL.h"
+
+/**
+ * @addtogroup base
+ * @{
+ */
+NS_CC_BEGIN
+
+class CC_DLL StencilBitsManager
+{
+public:
+    static StencilBitsManager* getInstance()
+    {
+        if(nullptr == _stencilBitsManager)
+        {
+            _stencilBitsManager = new StencilBitsManager();
+        }
+        return _stencilBitsManager;
+    }
+    GLint getStencilLayerMask()const;
+    void setStencilLayerMask(GLint stencilLayer);
+private:
+    static StencilBitsManager* _stencilBitsManager;
+    StencilBitsManager();
+    StencilBitsManager(const StencilBitsManager& other);
+    StencilBitsManager& operator=(const StencilBitsManager& other);
+    GLint _stencilLayerMask;
+};
+
+NS_CC_END
+// end of base group
+/** @} */
+#endif /* StencilBitsManager_hpp */
```

Comment:
My suggestion is that we can refine it. The `StencilBitsManager ` can act like a `stencil bits` pool, every one want `stencil bits` must try to get it, when it finished using it, the `stencil bits` must be returned to the `StencilBitsManager`.